packages/ui/src/utilities/copyDataFromLocale.ts

```diff
@@ -70,8 +71,9 @@ function iterateFields(
           // if the field has no value, take the source value
           if (
             field.name in toLocaleData &&
-            // only replace if the target value is null or undefined
-            [null, undefined].includes(toLocaleData[field.name]) &&
+            // only replace if the target value is null, undefined, or empty array
+            ([null, undefined].includes(toLocaleData[field.name]) ||
+              (Array.isArray(toLocaleData[field.name]) && toLocaleData[field.name].length === 0)) &&
```

Comment:
MongoDB and PostgreSQL behave differently when querying documents that don't exist in a specific locale:
- PostgreSQL: Returns the document with data from the fallback locale
- MongoDB: Returns the document with empty arrays for localized fields

This difference caused `copyToLocale` to fail in MongoDB because the merge algorithm only checked for `null` or `undefined` values, but not empty arrays. When MongoDB returned `content: []` for a non-existent locale, the algorithm would attempt to iterate over the empty array instead of using the source locale's data.